### PR TITLE
contrib/grpc: attempt to fix flaky tests

### PR DESCRIPTION
**Pull request by RomainMuller**
### What does this PR do?

- AppSec tests are occasionally receiving an unexpected `io.EOF` error from a gRPC stream send, which according to the documentation happens if the error does not originate from the client (and in those cases, blocking causes the error server-side)
- Some tests were locally flaking due to expecting a certain count of security events being discovered, but this sometimes was not met due to the WAF timeout (this likely would not happen in CI, where we seront a higher-than-default timeout via environment)
- Some other (non-AppSec) tests fail (possibly due to AppSec being there), but would not reproduce locally... so this PR replaces some calls to `require` with calls to `assert` to allow collection of more information about what's happened; and adds some new `assert` checks at key locations where errors could have been silently ignored previously.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Review by rarguelloF — APPROVED** [review]

File: contrib/google.golang.org/grpc/appsec_test.go [hunk 1]
```diff
@@ -24,26 +28,26 @@ import (
 )
 
 func TestAppSec(t *testing.T) {
-	appsec.Start()
+	appsec.Start(appsecConfig.WithWAFTimeout(time.Hour /* functionally unlimited */))
```
Comment (darccio) [inline contrib/google.golang.org/grpc/appsec_test.go:31]: @RomainMuller If `WithWAFTimeout` is only for this, could we just set the environment variable?

```suggestion
	t.SetEnv("DD_APPSEC_WAF_TIMEOUT", "1h")
	appsec.Start()
```

**Pull request merged**